include/boost/gil/detail/math.hpp

```diff
@@ -28,6 +43,124 @@ inline detail::kernel_2d<T, Allocator> get_identity_kernel()
     return kernel;
 }
 
+// Please refer https://stackoverflow.com/a/10032882/14958679 for getting an overview of the
+// concept applied for obtaining higher order sobel kernels.
+
+/// \defgroup KernelGeneration
+/// \brief Contains documentation for functions used in kernel generation.
+///
+/// Separate functions are used for generating only those kernels whose dimensions are greater than
+/// 5x5. Smaller kernels are fed directly to the algorithm.
+///
+
+/// \addtogroup KernelGeneration
+/// @{
+
+/// \brief Produces higher order kernel vector by performing discrete convolution between lower
+/// order kernel vector and a smoothing kernel vector(kernel used for suppressing noise).
+/// \param kernel1 - First argument for kernel vector convolution.
+/// \param kernel2 - Second argument for kernel vector convolution.
+/// \tparam T1 - Type of first argument for kernel vector convolution.
+/// \tparam T2 - Type of second argument for kernel vector convolution.
+template<typename T1, typename T2>
+std::vector<std::vector<float>> kernel_convolve_impl(T1 kernel1, T2 kernel2)
+{
+    size_t convolved_kernel_size = kernel1.size() + kernel2.size() - 1;
+    std::vector<std::vector<float>> convolved_kernel(convolved_kernel_size,
+                                                         std::vector<float>(convolved_kernel_size)),
+                                    dummy_kernel(convolved_kernel_size,
+                                                         std::vector<float>(convolved_kernel_size));
+    
+    // 'dummy_kernel' will be made by padding 'kernel1' with appropriate no. of rows and columns 
+    // containing zeros to match the size specified by 'convolved_kernel_size'.
+    // 'convolved kernel' will store the result obtained after applying convolution between 
+    // 'dummy_kernel' and 'kernel2'.
+    // 'padding_reference' will be used to apply appropriate padding with zeros around 'kernel1'
+    // to create 'dummy_kernel'.
+    
+    std::ptrdiff_t padding_reference = (kernel2.size() - 1) / 2;
+    for(std::ptrdiff_t row = padding_reference;row < dummy_kernel.size() - padding_reference; ++row)
+        for(std::ptrdiff_t col = padding_reference;col < dummy_kernel.size() - padding_reference;
+                                                                                            ++col)
+        {
+            dummy_kernel[row][col] = kernel1[row - padding_reference][col - padding_reference];
+        }
+
+    std::ptrdiff_t flip_ker_row, flip_ker_col, row_boundary, col_boundary;
```

Comment:
`s/ker/kernel/`

Reply:
I am not sure if I understood this correctly, shall I rename ```flip_ker_row``` to ```flip_kernel_row```?

Reply:
Yes, please - full words are preferred